posix/posix.c

```diff
@@ -1715,14 +1688,16 @@ int posix_ioctl(int fildes, unsigned long request, char *ustack)
 			default:
 				if (((request & IOC_INOUT) != 0) || (IOCPARM_LEN(request) > 0)) {
 					GETFROMSTACK(ustack, void *, data, 2);
+					GETFROMSTACK(ustack, size_t, size, 3);
```

Comment:
As we have the size of `data` now, `data` pointer can be checked with `vm_mapBelongs()` (see other syscalls for reference). This is an optional syscall parameter, so it should probably be done here - in this case include a comment in `syscalls_sys_ioctl()` so that it is apparent you took care of securing this syscall.